CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
+  * **fix** for bug fixes.
+  * **build** for build or CI-related improvements.
+  * **docs** for enhancements to documentation (only).
+  * **refactor** for refactoring (no logic changes).
+
+* PR titles also take an *optional scope* field containing the area
```

Comment:
Should we list some of the optional scope fields for usage consistency? Can we copy the github tags?

Reply:
Good idea. What tags should we add? Here are some initial ideas:

> vectors, types, expression-eval, operators, memory, dwio, parquet, dwrf, filesystem

Reply:
`fix` and `feat` are specific to changes that would require a bump of the patch or minor component of the version. So there could be rare cases where we have a `feat(build):` maybe adding a cmake config file or maybe removing a previously available option  but usually cmake changes would be `build:` because they don't influence the actual API of the package (which ~= the way velox is consumed [in binary form].). Though 'in binary form' might only be accurate once we have a proper release, because currently everyone uses velox from source?